Sources/SwiftParser/TokenSpecSet.swift

```diff
@@ -478,48 +481,88 @@ enum Operator: TokenSpecSet {
   }
 }
 
-/// Tokens that can be used in operator declarations
-enum OperatorLike: TokenSpecSet {
-  case `operator`(Operator)
-  case exclamationMark
+/// Tokens that are either binary operators, or can act like binary operators.
+enum BinaryOperatorLike: TokenSpecSet {
+  case binaryOperator
   case infixQuestionMark
-  case postfixQuestionMark
   case equal
   case arrow
 
   init?(lexeme: Lexer.Lexeme) {
-    if let op = Operator(lexeme: lexeme) {
-      self = .operator(op)
-      return
-    }
     switch lexeme.rawTokenKind {
-    case .exclamationMark: self = .exclamationMark
+    case .binaryOperator: self = .binaryOperator
     case .infixQuestionMark: self = .infixQuestionMark
-    case .postfixQuestionMark: self = .postfixQuestionMark
     case .equal: self = .equal
     case .arrow: self = .arrow
     default: return nil
     }
   }
 
-  static var allCases: [OperatorLike] {
-    return Operator.allCases.map(Self.operator) + [
-      .exclamationMark,
-      .infixQuestionMark,
-      .postfixQuestionMark,
-      .equal,
-      .arrow,
-    ]
+  var spec: TokenSpec {
+    switch self {
+    case .binaryOperator: return .binaryOperator
+    case .infixQuestionMark: return TokenSpec(.infixQuestionMark, remapping: .binaryOperator)
+    case .equal: return TokenSpec(.equal, remapping: .binaryOperator)
+    case .arrow: return TokenSpec(.arrow, remapping: .binaryOperator)
+    }
+  }
+}
+
+/// Tokens that are either postfix operators, or can act like postfix operators.
+enum PostfixOperatorLike: TokenSpecSet {
+  case postfixOperator
+  case exclamationMark
+  case postfixQuestionMark
+
+  init?(lexeme: Lexer.Lexeme) {
+    switch lexeme.rawTokenKind {
+    case .postfixOperator: self = .postfixOperator
+    case .exclamationMark: self = .exclamationMark
+    case .postfixQuestionMark: self = .postfixQuestionMark
+    default: return nil
+    }
   }
 
   var spec: TokenSpec {
     switch self {
-    case .operator(let op): return op.spec
+    case .postfixOperator: return .postfixOperator
     case .exclamationMark: return TokenSpec(.exclamationMark, remapping: .postfixOperator)
-    case .infixQuestionMark: return TokenSpec(.infixQuestionMark, remapping: .binaryOperator)
     case .postfixQuestionMark: return TokenSpec(.postfixQuestionMark, remapping: .postfixOperator)
-    case .equal: return TokenSpec(.equal, remapping: .binaryOperator)
-    case .arrow: return TokenSpec(.arrow, remapping: .binaryOperator)
+    }
+  }
+}
+
+/// Tokens that can be used in operator declarations
+enum OperatorLike: TokenSpecSet {
+  case `operator`(Operator)
```

Comment:
`binaryOperator` and `postfixOperator` should be covered by `BinaryOperatorLike` and `PostfixOperatorLike` resp. So, I think you can simplify this case to

```suggestion
  case prefixOperator
```